### PR TITLE
Decomposition for torch.aten.index_select

### PR DESCRIPTION
**Pull request by flemairen6**

### CODE REVIEW

**Review by ttjost — APPROVED** [review]

**Review by mgehre-amd — APPROVED** [review]
LGTM!
Love it when bugs turn out to be easy :-)

**Comment by flemairen6** [comment]
> LGTM! Love it when bugs turn out to be easy :-)

It's not THAT easy, decomposing the operator generates an `index.Tensor` that is not lowered right now - so unfortunately, it will require a bit more work (but this can be merged independently)

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -3784,34 +3797,36 @@ LogicalResult ConvertAtenOp<Aten_IndexPutImplOp>::matchAndRewrite(
...
 
-
   auto numInElements = self.getType().getShape()[0];
   auto numValues = values.getType().getShape()[0];
 
   // TOSA scatter requires 3d in and 2d indices & values
-  SmallVector<int64_t> scatterInOutShape {1, numInElements, 1};
-  SmallVector<int64_t> scatterIndicesShape {1, numValues};
-  SmallVector<int64_t> scatterInputShape {1, numValues, 1};
+  SmallVector<int64_t> scatterInOutShape{1, numInElements, 1};
```
Comment (flemairen6) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:3810]: Did you run some kind of formatter? It feels like there are a lot of those changes

File: include/torch-mlir/Conversion/TorchToTosa/TosaLegalizeUtils.h [hunk 2]
```diff
@@ -123,6 +123,10 @@ void CreateReplaceOpAndInfer(PatternRewriter &rewriter, Operation *op,
 TypedValue<RankedTensorType> reshapeTo(Location loc, PatternRewriter &rewriter,
                                        Value val, ArrayRef<int64_t> newShape);
 
+TypedValue<RankedTensorType> tranposeBy(Location loc,
```
Comment (ttjost) [inline include/torch-mlir/Conversion/TorchToTosa/TosaLegalizeUtils.h:126]:
```suggestion
TypedValue<RankedTensorType> transposeBy(Location loc,
```

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 1 above)
Comment (mgehre-amd) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:3810]: Hm, it was supposed to only format the lines I modified. I will revert the rest.

**Pull request merged**